workflow/scripts/osemosys_global/powerplant/renewable_targets.py

```diff
@@ -0,0 +1,212 @@
+                    list(itertools.product(target + node_dict[target], target_years)), 
+                    columns=["FUEL", "YEAR"]
+                )
+                re_targets_df["GEO"] = node_dict[target]
+                re_targets_df = re_targets_df[["GEO", "YEAR"]]
+                
+                re_targets_df['VALUE'] = value_dict[target] / 100
+                re_targets_df.drop_duplicates(inplace=True)
+    
+                re_targets_df = re_targets_df.pivot(
+                    index=["YEAR"], columns=["GEO"], values="VALUE"
+                ).reset_index()
+                re_targets_df = re_targets_df.interpolate()
+        
+                # Drop all columns with only NaN
+                re_targets_df.dropna(axis=1, how="all", inplace=True)
+        
+                # Melt to get 'COUNTRY' column and remove all rows with NaN
+                re_targets_df = pd.melt(
+                    re_targets_df,
+                    id_vars=["YEAR"],
+                    value_vars=[x for x in re_targets_df.columns if x not in ["YEAR"]],
+                    var_name="GEO",
+                    value_name="VALUE",
+                )
+                re_targets_df.dropna(axis=0, inplace=True)
+        
+                # Read 'SpecifiedAnnualDemand' as output from the demand_projection rule.
+                sp_demand_df = annual_demand_df.loc[
+                    ~(annual_demand_df["FUEL"].str[3:8].isin(remove_nodes))
+                ]
+                  
+                if len(str(node_dict[target])) == 3:
+                    sp_demand_df["GEO"] = sp_demand_df["FUEL"].str[3:6]
+                    
+                elif len(str(node_dict[target])) == 5:
+                    sp_demand_df["GEO"] = sp_demand_df["FUEL"].str[3:8]
+                    
+                sp_demand_df = sp_demand_df.groupby(["YEAR", "GEO"], as_index=False)[
+                    "VALUE"
+                ].sum()
+                sp_demand_df.rename(columns={"VALUE": "DEMAND"}, inplace=True)
+                
+                # Merge RE targets and SpecifiedAnnualDemand tables
+                re_targets_df = pd.merge(
+                    re_targets_df, sp_demand_df, how="left", on=["GEO", "YEAR"]
+                )
+                
+                # Calculate the country level target for the dummy RES commodity
+                re_targets_df["VALUE"] = re_targets_df["VALUE"] * re_targets_df["DEMAND"]
+                re_targets_df["FUEL"] = target + re_targets_df["GEO"]
+                re_targets_df["REGION"] = region_name
+                re_targets_df = re_targets_df[["REGION", "FUEL", "YEAR", "VALUE"]]
+                
+                if accumulated_annual_demand.empty:
+                
+                    accumulated_annual_demand = re_targets_df
+                
+                else:
+                    accumulated_annual_demand = pd.concat([
+                        accumulated_annual_demand, 
+                        re_targets_df])
```

Comment:
Similar to a previous comment (and not a required change) but its usually better to do `pd.concat(...)` once outside the loop. ie. Collect all dataframes in a list instead 

````
dfs = []
for x in data_list:
    # do work 
    df = pd.DataFrame()
    dfs.append(df)

final = pd.concat(dfs)
```